### DISCUSSION TITLE
Nieoczekiwane płatności

### DISCUSSION THREAD

**Discussion by vetomir** [AzureCommunityPL/group#157] [Pomoc] [answered]
Hejże, jakiś czas temu podłączyłem się do Azure i zacząłem się uczyć platformy. Głównie używałem App Services, Sql database i Face Api do Javy (Spring Boota), po tych 30 dniach przeszedłem na "Subskrybcję platformy Azure", gdzie cały czas zostawałem zasypywany informacjami że jest to nadal usługa darmowa, ale i tak co jakiś czas sprawdzałem saldo rachunków gdzie widniało okrągłe 0 Euro. 

Dzisiaj z kolei dostałem maila w którym dostałem ponaglenia do zapłaty w wysokości 600 Euro, co wbiło mnie w podłogę i nie mam pojęcia skąd to się wzięło. Napisałem oczywiście zażalenie do supportu i prośbę o zgłoszenie się do mnie w tej sprawie, gdzie oczywiście dodałem że się dopiero uczę, nie miałem pojęcia itd. 

Spotkał się już może ktoś z czymś takim? I czego się mogę spodziewać?

Pozdrawiam Wszystkich

**Comment by vetomir** [comment]
![image](https://user-images.githubusercontent.com/22453248/99992484-b0e49400-2db6-11eb-88c8-d7683478f537.png)
To wszystko minus te początkowe środki

**Comment by kamil-mrzyglod** [comment]
No dobra, a ile tych baz masz i jaki to tier? To samo z App Service - tam przypadkiem nie było tak, że po 30 dniach darmowy jest tylko F1 tier? Patrzę generalnie na listę tych darmowych usług i taki PostgreSQL na pewno tam nie widnieje 😛 

https://azure.microsoft.com/en-us/free/free-account-faq/

**Reply by vetomir** [reply]
PostgreSQL wrzuciłem jeszcze na tym darmowym, z resztą jak zobaczyłem ile zjadło usunąłem w 5 minut.
Co do reszty to tak jak było w domyślnych tak dałem, wraz z tym rachunkiem wywaliłem wszystko, więc nie za bardzo wiem jak sprawdzić.

**Reply by kamil-mrzyglod** [reply]
Wiesz, generalnie jest tak, że w tych 30 dniach miałeś do wydania 200$ i to funkcjonuje jako tzw. spending limit czyli możesz maksymalnie tyle wydać. Jak w ciągu 30 dni dobijesz do limitu to albo robisz upgrade do PAUG, albo zasoby są blokowane i możesz kiedyś wrócić do tematu aby je dalej używać.

Jak przekroczyłeś 30 dni i zrobiłeś upgrade do PAUG to generalnie masz dostęp do darmowych i płatnych usług z tą różnicą, że w darmowych musisz sie poruszać według limitów. Jeśli masz 750h dla konkretnej rodziny VMek to tego nie możesz przekroczyć(czyli możesz mieć kilka VM, ale całkowite zużycie nie może przekroczyć limitu). 

Z drugiej strony takiego Azure App Service nie masz wylistowanego w ramach puli darmowych usług w ciągu 12 miesięcy więc jak postawiłeś sobie coś na tej usłudze używając innego tier niż darmowy, to po upgrade do PAUG zaczęły się naliczać koszty. Dla SQL Database masz powiedziane, że masz dostęp każdego miesiąca do bazy S0 z 250GB i 10 DTU. Jak coś pokręciłeś i skorzystałeś z innego tier(bądź stworzyłeś kilka instancji) to całkiem możliwe, że zaczęło się naliczanie.

Generalnie powinni Ci ubić większość jak nie całość kosztów jak wykonasz czynny żal. Ja zawsze leciałem na subskrypcjach z limitem(typu MSDN) albo PAUG więc nie mam za dużo doświadczenia z przekraczaniem tych magicznych 30 dni. Z drugiej strony nawet mając spending limit możesz się naciąć na koszt, który pod niego nie podlega(swego czasu tak było z kosztem za Application Insights jak przekroczyłeś darmowe GB na metryki).

Pytanie jeszcze kiedy skończyłeś się upewniać ile co kosztuje? Billing ma to do siebie, że ma trochę opóźnienia więc mogłeś się nie zorientować, ale ten Free Account ma właśnie ten urok, że okres ochronny kończy się po 30 dniach. Potem można się dość łatwo naciąć.

**Reply by eklime** [reply]
No postgresql nie zjada w 5 minut prawie 150 eurasow...
Na poziomie subskrypcji powinieneś mieć widok na koszty - rozbij sobie na dni i grupowanie po resource i zobacz które konkretnie usługi Ci żarły.
Możesz też wyeksportować sobie z cost management do CSV zestawienie to zobaczczysz jakie miałeś rozmiary i ile jednostek ci zeżarło.

**Reply by kamil-mrzyglod** [reply]
No SQL Database to też musiał być jakiś napuchnięty bo nawet używając S0 masz te 14 EUR miesięcznie zakładając rozliczenie DTU. Skoro Tobie natłukło ponad 600 to musiałeś nieźle szaleć albo się władować w vCore :)

**Comment by vetomir** [comment]
@kamil-mrzyglod Dzięki wielkie, konto mam od czerwca i ostatni raz chyba tam zajrzałem z tydzień temu i było wciąż pusto. Zobaczę co powiedzą z supportu.

@eklime Co do Postgresa jestem pewny że go usunąłem kilka minut po starcie, tą kwotę naliczyło w momencie założenia, ale wtedy był mój pierwszy kontakt z chmurą i mogłem coś w ustawieniach poprzestawiać.

**Reply by kamil-mrzyglod** [reply]
To zobacz jeszcze tak jak Emil pisał - wejdź w subskrypcję i wygeneruj sobie raport z ostatnich miesięcy. To powinno dać lepszy obraz skąd takie zużycie.

**Reply by vetomir** [reply]
Widzę z tego wygląda, że zjadało mi od miesiąca po 30-40 Euro dziennie, a wcześniej drobne kwoty, na dodatek baza która miała z 10 recordów naliczała tyle samo co ta z kilkoma tysiącami, nie wiem gdzie jest ta darmowość .Także prawdopodobnie po prostu cały czas źle patrzyłem. No nic, będę grał wariata, chyba tylko coś takiego mi zostało.

**Comment by vetomir** [comment, marked as answer]
Więc, tak. Bardzo dziękuję za Waszą pomoc, udało mi się wynegocjować dużo niższą stawkę z suportem Microsoftu, dostałem również wszystkie zalecenia, wraz z kalkulatorami które pozwalają mi na darmowe użytkowanie platformy.

Pozdrawiam!
